### PR TITLE
Update search block

### LINKED ISSUE #643: Search block with button icon does not look the same in the editor and front
**Describe the bug**
When a search block has the option "use button with icon" enabled, the editor and front does not match.

The colors when the button is hovered are also different.

**To Reproduce**
1. Add a search block.
2. Enabled the option to display the button icon
3. Compare the editor and front

Next, repeat this in combination with the other settings: Button inside, button only.

**Expected behavior**
The editor and front should match.

**Screenshots**
Left: Editor. Right: Front.

Button only:
searchblock-buttononly-icon

Button inside:
searchblock-buttoninside-icon

Button outside:
searchblock-buttonoutside-icon

**Editor version (please complete the following information):**
5.6-beta1-49267

### PR DESCRIPTION
**Pull request by carolinan**
Fixes https://github.com/WordPress/twentytwentyone/issues/643

## Relevant technical choices:

Increased icon size to 40px.
Adjusted padding.
Adjusted front facing hover style.

Adjusted colors for when the block is placed in a block with a background.

-The front is 1px off compared to the editor 🤔 

## Test instructions

This PR can be tested by following these steps:
1. Add a search block
1. compare the block styles in the editor and front.
1.

## Screenshots

Editor:

search-editor

Front:
search-front

Front, hover:
search-front-hover

## Quality assurance
* [x] I have thought about any security implications this code might add.
* [x] I have checked that this code doesn't impact performance (greatly).
* [x] I have tested this code to the best of my abilities
* [x] I have checked that this code does not affect the accessibility negatively

### CODE REVIEW

**Comment by carolinan** [comment]
I am not sure how to solve the hover colors for when the block is placed inside a block with a background color.

**Comment by melchoyce** [comment]
Hmm yeah it's weird that the search icon is centered in the editor, but off on the front-end 🤔 @ryelle or @jffng could you take a look?

> I am not sure how to solve the hover colors for when the block is placed inside a block with a background color.

I wonder if there's someone on the Gutenberg team we could ping to chat about this with. @karmatosed or @jasmussen, do you know?

**Comment by carolinan** [comment]
The icon size/padding/the ghost that is messing up the styles is changing the height of the input field so that the height in the editor and front doesn't match.

**Comment by jffng** [comment]
> Hmm yeah it's weird that the search icon is centered in the editor, but off on the front-end 🤔 @ryelle or @jffng could you take a look?

On the front-end, the icon button is inheriting a 1.5 line-height from the shared button styles:  

Screen Shot 2020-10-23 at 1 44 42 PM

If you override it / set it to 1, the icon shows up centered for me:

Screen Shot 2020-10-23 at 1 42 38 PM

> the ghost that is messing up the styles is changing the height of the input field

👻 😆

**Comment by jasmussen** [comment]
Nice sleuthing, Jeff.

If you feel like the search block should account for that inheritance directly in the block (seems to me it should), I can help shepard that PR along. Our I can take a stab Monday.

**Comment by carolinan** [comment]
> Nice sleuthing, Jeff.
> 
> If you feel like the search block should account for that inheritance directly in the block (seems to me it should), I can help shepard that PR along. Our I can take a stab Monday.

I don't speak Gutenberg, if anyone wants to translate...

**Comment by carolinan** [comment]
> > I am not sure how to solve the hover colors for when the block is placed inside a block with a background color.
> 
> I wonder if there's someone on the Gutenberg team we could ping to chat about this with. @karmatosed or @jasmussen, do you know?

This is because of the themes color setting where the text color changes with the **body** background color. Right now the blocks that has color options are all handled individually with different solutions 🤦

**Comment by jasmussen** [comment]
> I don't speak Gutenberg, if anyone wants to translate...

Sure, [here is the PR I was referring to](https://github.com/WordPress/gutenberg/pull/26446), that way in the future, the centering issue is fixed for all themes.

**Comment by melchoyce** [comment]
With the line-height update in, does this need anything else? The original screenshots look good otherwise.

**Comment by carolinan** [comment]
Something is causing problems with styles of the input fields when placed inside a dark group block.
It does not look the same now as it did when I worked on https://github.com/WordPress/twentytwentyone/pull/673/commits/9b18e9cda8540dc6989eb29ac436992fa1d04f6d so I don't know what is going on.

Possible an issue with beta 5.6-beta1-49308

**Comment by carolinan** [comment]
This is not right, trying again.
border-problems

**Comment by carolinan** [comment]
These merge conflicts are beyond repair because of the changes to the buttons, keeping it open just for reference while I try again... 😿

**Comment by carolinan** [comment]
New test content: https://gist.github.com/carolinan/0e5c4337ac75398327430f435a0fda90

**Comment by carolinan** [comment]
https://github.com/WordPress/twentytwentyone/pull/673/commits/e4558d9720f56e4d577f94864334a4499f124f28 Updates colors, and the size of the button when it is placed inside.
It also removes the duplicate borders on the input field that were visible in the editor.
 
<del>**It is not ready to be merged** because there are plenty of problems when the body background color is changed 😫 </del>

On closer testing, the border of the input field does not change when in the customizer, and this is a separate issue.
-I also noticed that the vertical distance between the label and the form seems to have been reduced at some point.

This PR is ready for review.

(There is a little bit of bleed in the screenshot because I reduced its size)
search-block

**Review by kjellr — COMMENTED** [review]
In general this looks good. The only issue I'm still seeing is that against some of the theme's lighter background colors, things look differently in the editor than they do on the front end: 

Editor|Front End
---|---
Screen Shot 2020-10-29 at 1 44 12 PM|Screen Shot 2020-10-29 at 1 44 07 PM

The front-end is correct there — is it possible to get those synced up?

**Comment by carolinan** [comment]
On the front, the border is white for me with these colors, not black.
For every change we do, 4 new problems are found :/

The background color is also the body background color, not the "parent" block background color, as the screenshot above shows.

**Comment by kjellr** [comment]
When re-testing now, I see this in both the editor and the front end. It's a little different from what I saw last night, so not sure what that's about. 

Before|After
---|---
Screen Shot 2020-10-30 at 9 16 59 AM|Screen Shot 2020-10-30 at 9 18 39 AM

File: assets/sass/05-blocks/search/_editor.scss [hunk 1]
```diff
@@ -33,6 +46,75 @@
...
+				width: 40px;
+				height: 40px;
+			}
+		}
+
+		.has-background:not(.has-background-background-color) &,
+		[class*="background-color"]:not(.has-background-background-color) &,
+		[style*="background-color"]:not(.has-background-background-color) &,
+		.wp-block-cover[style*="background-image"] & {
+			color: var(--global--color-white);
```
Comment (kjellr) [inline assets/sass/05-blocks/search/_editor.scss:68]: Shouldn't the color here be changed only for darker backgrounds? I think this color rule should apply to the same selectors from L31-33 above: 

```
.has-background.has-gray-background-color &,
.has-background.has-dark-gray-background-color &,
.has-background.has-black-background-color &
```

**Comment by carolinan** [comment]
Front,
Black body background, group block with purple background:
hover

Editor:
Black body background, the search block is not inside another block.
Here, the styles are not working when the button is inside.
_This gif is out of date :)_

editor-hover

**Comment by hedgefield** [comment]
Tested the changes on my staging site, looks good to me!

**Pull request merged**